src/entities/createPool/createPoolV3/weighted/createPoolWeighted.ts

```diff
@@ -0,0 +1,64 @@
+import { getRandomBytes32 } from '@/entities/utils/getRandomBytes32';
+import { encodeFunctionData } from 'viem';
+import {
+    CreatePoolBase,
+    CreatePoolV3WeightedInput,
+    CreatePoolBuildCallOutput,
+    CreatePoolV3WeightedArgs,
+    TokenConfig,
+} from '../../types';
+import { weightedPoolFactoryV3Abi } from '@/abi/weightedPoolFactory.V3';
+
+export class CreatePoolWeightedV3 implements CreatePoolBase {
+    buildCall(input: CreatePoolV3WeightedInput): CreatePoolBuildCallOutput {
+        const args = this.parseCreateFunctionArgs(input);
+        const encodedCall = encodeFunctionData({
+            abi: weightedPoolFactoryV3Abi,
+            functionName: 'create',
+            args,
+        });
+        return { call: encodedCall };
+    }
+
+    private parseCreateFunctionArgs(
+        input: CreatePoolV3WeightedInput,
+    ): CreatePoolV3WeightedArgs {
+        const sortedTokenParams = input.tokens.sort(
```

Comment:
I wonder if something along the lines of [this](https://github.com/balancer/balancer-v3-monorepo/blob/9d1dd909555309f5c59a504c2dca7fc6018fea8a/pvt/helpers/src/models/tokens/ERC20TokenList.ts#L168) is easier to follow? 
We can probably break the sorting to a separate helper function too and re-use as the same is being done in each pool type creation.

Reply:
Actually I already did a function to sort tokens by address, I forgot to reuse 😅